explorer/src/commitUtils.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+type CommitData = {
+  // TODO improve variable names
+  fileToCommits: {[filename: string]: string[]};
+  commits: {[commithash: string]: Commit};
```

Comment:
Do you know why we get `no-use-before-define` lints here and in similar places (lines 12, 34)? Pasting these three paragraphs into https://flow.org/try yields no errors ([direct link here](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgDCcAtiQJYYAiAhhjWALxgDeqYYwwYAKgPJU+YciRwAnOADcCkmmPI0ARjAIA7GiTwBndmCjkVPOMTKUtALlYBtfSvWbLWjPNUBzALqPn5N1fcBfAG5dAGNSCgwLazDTDAALGi04rxcPSxMIoN0aAFd4uDEop1S-YP90bHwicMomVmy8uIKUn1dgjic6KJYbAzx7PBa3TzAAMT6AZXpIrPLMXAJxlSmuurYOGB9tS1UckkU8MXawGgATU7xTnb2Do90LlQxL6-3DstQgA)).

I'd love to not get into the habit of ignoring lint at this stage. :-)